packages/gatsby-theme-store/src/sdk/search/useSearchInfinite.ts

```diff
@@ -48,46 +48,26 @@ export const useSearchInfinite = <Query extends BaseQueryShape | undefined>({
       // This is a pre-rendered search. Like so, we need to fetch the data
       // at the exact same order from the pre-rendered data so we don't have
       // data mismatch
-      // TODO: This breaks regionalization, so we don't do it if regionId is set.
-      // Need to look into it.
-      if (!regionId) {
```

Comment:
removing this check breaks regionalization if the products from the chosen region differ from the ones on the pre-rendered page